app.js

```diff
@@ -136,6 +137,32 @@ function addEventListener(indicator, emitter, event, f) {
   });
 }
 
+/*
+  Check the referenced database exists and that the credentials used have
+  visibility. Callback with a fatal error if there is a problem with the DB.
+  @param {string} db - database object
+  @param {function(err)} callback - error is undefined if DB exists
+*/
+function proceedIfDbValid(db, callback) {
+  db.head('', function(err) {
+    err = error.convertResponseError(err, function(err) {
+      if (err && err.statusCode === 404) {
+        // Override the error type and mesasge for the DB not found case
+        var msg = `Database ${db.config.url}/${db.config.db} does not exist. ` +
```

Comment:
We're possibly leaking creds here. Maybe do `db.config.url.replace(/\/\/.+@/g, '//****:****@')`?

Reply:
The message is identical to what it was previously, but it's a fair point, I'll sort it out.

Reply:
Done in 9f6a347a0fbd9290cac47cb82fa8f4e080ba9188